src/http/mod.rs

```diff
@@ -28,7 +39,18 @@ pub trait RequestExt<'a>: Sized {
     fn send_with_client_config<B: IntoRequestBody + 'a>(self, body: B, client_config: Arc<ClientConfig>) -> RequestSend<'a>;
 }
 
-impl<'a> RequestExt<'a> for &'a http::Request<()> {
+impl<'a,T: IntoNonUnitRequestBody + Clone> RequestWithBodyExt<'a> for &'a http::Request<T> {
+    type B = T;
+    fn send_with_client_config(self, client_config: Arc<ClientConfig>) -> RequestSend<'a> {
+	let cloned_self = (*self).clone();
+        let (read, len) = cloned_self.into_body().into_request_body();
```

Comment:
doesn't this make the `clone()` unnecessary?:
```rust
   let (head, body) = self.into_parts();
   let req = http::Request::from_parts(head, ());
   req.send_with_client_config(req, body, client_config)
```

Then we can also drop the additional type argument in `request_native.rs::RequestSend::new_with_client_config` and the `Clone` requirement here.

Reply:
we implement `RequestWithbodyExt` for &'a Request<T> consuming self (when using `into_parts`) seems to be a problem.